org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Nisida/ImportNisida.java

```diff
@@ -0,0 +1,576 @@
+	 * 
+	 * @param value  string to be parsed
+	 * @param status supporter object
+	 * @return double value, or null if field is empty
+	 */
+	public static Double valueFor(final String value, final NisidaLoadState status) {
+		final Double res;
+		if (value.length() == 0) {
+			res = null;
+		} else if (value.equals("-")) {
+			res = null;
+		} else {
+			try {
+				res = Double.parseDouble(value);
+			} catch (NumberFormatException nfe) {
+				status.getErrors().add(new ImportNisidaError("Error on line " + status.getLineNumber(),
+						"Failed to parse numeric field - " + value));
+				return null;
+			}
+		}
+		return res;
+	}
+
+	private static void processPosition(final String[] tokens, final NisidaLoadState status) {
+		// sample:
+		// 311002Z/3623.00N/00412.02E/GPS/359/03/-/
+		final HiResDate dtg = new HiResDate(status.timestamp);
+		final Double latVal = parseDegrees(tokens[1], status);
+		final Double longVal = parseDegrees(tokens[2], status);
+		if (dtg != null && latVal != null && longVal != null) {
+			final WorldLocation location = new WorldLocation(latVal, longVal, 0d);
+
+			final String source = tokens[3];
+			final Double courseVal = valueFor(tokens[4], status);
+			final Double speedVal = valueFor(tokens[5], status);
+			final Double depthVal = valueFor(tokens[6], status);
+
+			Fix fix = new Fix(dtg, location, 0d, 0d);
+			if (courseVal != null) {
+				fix.setCourse(Math.toRadians(courseVal));
+			}
+			if (speedVal != null) {
+				fix.setSpeed(Conversions.Kts2Yps(speedVal));
+			}
+			if (depthVal != null) {
+				fix.getLocation().setDepth(depthVal);
+			}
+			FixWrapper res = new FixWrapper(fix);
+
+			// sort out the sensor
+			final String sourceName = POS_SOURCE_TO_NAME.get(source);
+			if (sourceName != null) {
+				res.setLabel(sourceName);
+			} else {
+				res.setLabel(source);
+			}
+
+			status.getPlatform().addFix(res);
+		}
+	}
+
+	private void processEnvironment(Object dataStore, Object datafile, String changeId) {
+		// TODO Auto-generated method stub
+
+	}
+
+	private void processSensor(Object dataStore, Object datafile, String changeId) {
+		// TODO Auto-generated method stub
+
+	}
+
+	private void processMastexposure(Object dataStore, Object datafile, String changeId) {
+		// TODO Auto-generated method stub
+
+	}
+
+	private void processDipOrBoy(Object dataStore, Object datafile, String changeId) {
+		// TODO Auto-generated method stub
+
+	}
+
+	private void processAttack(Object dataStore, Object datafile, String changeId) {
+		// TODO Auto-generated method stub
+
+	}
+
+	private static void processDetection(final String[] tokens, final NisidaLoadState status) {
+		// sample:
+		// [DayTime/DET/DetectingSensor/Bearing/Range in NM/TN/Own Lat/Own Lon/Position Source/Remarks/]
+		// 311200Z/DET/RDR/23/20/777/3602.02N/00412.12E/GPS/DETECTION RECORD
+		
+
+		/**
+		 *  Create a sensor using the expanded `Sensor Code` name plus the track number (TN). 
```

Comment:
@saulhidalgoaular  - if we can't find the sensor code, don't worry about throwing an error. Just use the short token.

These files are human generated, and can contain lots of typos.  So we can be relaxed about things.

Reply:
Fixed at https://github.com/debrief/debrief/pull/4905/commits/b7787faa32aa4d068395ea853dae85884a2323e4